src/app/authoring-tool/edit-component-advanced/edit-component-advanced.component.ts

```diff
@@ -0,0 +1,27 @@
+import { Component, Inject, OnInit } from '@angular/core';
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material/dialog';
+import { Component as WiseComponent } from '../../../assets/wise5/common/Component';
+
+@Component({
+  selector: 'edit-component-advanced',
+  templateUrl: './edit-component-advanced.component.html',
+  styleUrls: ['./edit-component-advanced.component.scss']
+})
+export class EditComponentAdvancedComponent implements OnInit {
+  component: WiseComponent;
+  nodeId: string;
+
+  constructor(
+    public dialogRef: MatDialogRef<EditComponentAdvancedComponent>,
+    @Inject(MAT_DIALOG_DATA) public data
+  ) {
+    this.component = data.component;
+    this.nodeId = data.nodeId;
+  }
```

Comment:
We can declare the component variable in the constructor here? Since the nodeId is already in the WISEComponent, we don't need to pass it in, and instead get it via ```component.nodeId```?
```suggestion
  constructor(
    private dialogRef: MatDialogRef<EditComponentAdvancedComponent>,
    @Inject(MAT_DIALOG_DATA) protected component: WiseComponent
  ) {}
```